llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -2389,7 +2389,7 @@ class TargetLoweringBase {
                                                EVT VT) const {
     // If a target has multiple condition registers, then it likely has logical
     // operations on those registers.
-    if (hasMultipleConditionRegisters())
+    if (hasMultiplePredicateRegisters(VT))
```

Comment:
From looking at the use of this hook in `DAGCombiner::visitSELECT`, I think VT here is the type of X and Y. But `hasMultiplePredicateRegisters` should be passed the type of the predicate, which is hard coded to `MVT::i1`. (Please double check this since I find it quite confusing!)

Reply:
Of course it would help if the comment on this function said what `VT` is supposed to be.